File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) {
            return false;
        }

        long long originalNumber = x;
        long long reversedNumber = 0;

        while (x > 0) {
            int digit = x % 10;
            reversedNumber = reversedNumber * 10 + digit;
            x /= 10;
        }

        return originalNumber == reversedNumber;
    }
};

File: 0009-palindrome-number/0009-palindrome-number.rs
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        if x < 0 {
            return false;
        }

        let mut original = x;
        let mut reversed = 0;

        while original > 0 {
            reversed = reversed * 10 + original % 10;
            original /= 10;
        }

        x == reversed
    }
}